index.js

```diff
@@ -8,34 +8,44 @@
 (function () {
 	'use strict';
 
-	var hasOwn = {}.hasOwnProperty;
+	var hasOwnProperty = Object.prototype.hasOwnProperty;
+	var isArray = Array.isArray;
+	
+	function classNames() {
+		var len = arguments.length;
+
+		if (!len) return '';
+
+		var str = '', item, i, n;
+		for (i = 0; i < len; i++) {
+			if (!(item = arguments[i]))	continue;
+
+			if (typeof item === 'string' || typeof item === 'number') {
+				str && (str += ' ');
+				(str += item);
```

Comment:
Since we're chasing milliseconds, won't it be faster to just assign once?

```
str = (str ? ' ' : '') + n;
```

Reply:
Practically, for v8 my aproach is faster, for other browsers they're almost equal;
I suppose it is caused by prediction branching.
In your way you have 2 concats all the time, in mine - 1 guaranteed and one optional;

Reply:
Yes, but my gut feeling was that the extra concat done on item 0 is worth avoiding the extra variable mutations for items 1+.

Reply:
@krawaller benchmarks result:
mine:  `local#strings x 14,682,684 ops/sec ±1.21% (89 runs sampled)`
yours: `local#strings x 14,028,615 ops/sec ±1.01% (89 runs sampled)`

It is obviusly a very small difference but if it can be done - why not =)

Reply:
Yeah, I guess we've left the real world far behind. :)

Still, very interesting! I really thought variable mutation would be more expensive, but yours win out. Predictive branching for the win, I guess!

Reply:
@krawaller tbh first time i've seen the results i've been wondered too =)
First iteration code been written as yours %)